includes/Core/Consent_Mode/Consent_Mode.php

```diff
@@ -86,6 +88,66 @@ function () use ( $consent_mode_enabled ) {
 		);
 
 		add_filter( 'googlesitekit_inline_base_data', $this->get_method_proxy( 'inline_js_base_data' ) );
+
+		add_action( 'wp_ajax_install_activate_wp_consent_api', array( $this, 'install_activate_wp_consent_api' ) );
+	}
+
+	/**
+	 * AJAX callback that installs and activates the WP Consent API plugin.
+	 *
+	 * This function utilizes an AJAX approach instead of the standardized REST approach
+	 * due to the requirement of the Plugin_Upgrader class, which relies on functions
+	 * from `admin.php` among others. These functions are properly loaded during the
+	 * AJAX callback, ensuring the installation and activation processes can execute correctly.
+	 *
+	 * @since n.e.x.t
```

Comment:
I don't quite follow; what stops us from using our usual REST endpoint code/approach here instead of this approach?

Is the issue that we aren't loading the correct functions? Is there no bootstrap for this type of thing? How many files would need to be included to get things working? 🤔

Reply:
It requires using `admin.php` which is used for bootstrapping the page on load, including sending headers, etc. It is not recommended to use it in REST context, and it needs more files than that, which are loaded on page load, like wp version, etc. So we had to switch to AJAX approach